File: src/main/scala/org/openapitools/client/api/ShipmentInfoDTO.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime

import java.time.OffsetDateTime

import ShipmentInfoDTO._

case class ShipmentInfoDTO (
  /* Идентификатор отгрузки. */
  id: Option[Long],
/* Начало планового интервала отгрузки. */
  planIntervalFrom: Option[OffsetDateTime],
/* Конец планового интервала отгрузки. */
  planIntervalTo: Option[OffsetDateTime],
shipmentType: Option[ShipmentType],
warehouse: Option[PartnerShipmentWarehouseDTO],
warehouseTo: Option[PartnerShipmentWarehouseDTO],
/* Идентификатор отгрузки в вашей системе. Если вы еще не передавали идентификатор, вернется идентификатор из параметра `id`. */
  externalId: Option[String],
deliveryService: Option[DeliveryServiceDTO],
palletsCount: Option[PalletsCountDTO],
/* Идентификаторы заказов в отгрузке. */
  orderIds: Set[Long],
/* Количество заказов, которое Маркет запланировал к отгрузке. */
  draftCount: Option[Integer],
/* Количество заказов, которое Маркет подтвердил к отгрузке. */
  plannedCount: Option[Integer],
/* Количество заказов, принятых в сортировочном центре или пункте приема. */
  factCount: Option[Integer],
status: Option[ShipmentStatusType],
/* Описание статуса отгрузки. */
  statusDescription: Option[String],
/* Время последнего изменения статуса отгрузки. */
  statusUpdateTime: Option[OffsetDateTime])

object ShipmentInfoDTO {
  import DateTimeCodecs._

  implicit val ShipmentInfoDTOCodecJson: CodecJson[ShipmentInfoDTO] = CodecJson.derive[ShipmentInfoDTO]
  implicit val ShipmentInfoDTODecoder: EntityDecoder[ShipmentInfoDTO] = jsonOf[ShipmentInfoDTO]
  implicit val ShipmentInfoDTOEncoder: EntityEncoder[ShipmentInfoDTO] = jsonEncoderOf[ShipmentInfoDTO]
}

File: ympa_fsharp_giraffe_server/src/model/ShipmentInfoDTO.fs
namespace ympa_fsharp_giraffe_server.Model

open System
open System.Collections.Generic
open ympa_fsharp_giraffe_server.Model.DeliveryServiceDTO
open ympa_fsharp_giraffe_server.Model.PalletsCountDTO
open ympa_fsharp_giraffe_server.Model.PartnerShipmentWarehouseDTO
open ympa_fsharp_giraffe_server.Model.Set
open ympa_fsharp_giraffe_server.Model.ShipmentStatusType
open ympa_fsharp_giraffe_server.Model.ShipmentType

module ShipmentInfoDTO =

  //#region ShipmentInfoDTO


  type ShipmentInfoDTO = {
    Id : int64;
    PlanIntervalFrom : Nullable<DateTime>;
    PlanIntervalTo : Nullable<DateTime>;
    ShipmentType : ShipmentType;
    Warehouse : PartnerShipmentWarehouseDTO;
    WarehouseTo : PartnerShipmentWarehouseDTO;
    ExternalId : string;
    DeliveryService : DeliveryServiceDTO;
    PalletsCount : PalletsCountDTO;
    OrderIds : int64[];
    DraftCount : int;
    PlannedCount : int;
    FactCount : int;
    Status : ShipmentStatusType;
    StatusDescription : string;
    StatusUpdateTime : Nullable<DateTime>;
  }
  //#endregion
